### PR TITLE
Add Markdown Reference extension

### PR DESCRIPTION
**Pull request by jordobaron**
## Description

This extension allows you to quickly search Markdown syntax if you're like me and forget it a lot

## Type of change

New extension

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/markdown-reference/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  "name": "markdown-reference",
+  "title": "Markdown Reference",
+  "description": "Quickly find how to do something with markdown",
+  "icon": "logo.png",
+  "author": "codedbyjordan",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "markdown-reference",
+      "title": "Markdown Reference - Search Markdown Syntax",
```
Comment (pernielsentikaer) [inline extensions/markdown-reference/package.json:12]:
```suggestion
      "title": "Search Reference",
      "subtitle": "Markdown Syntax",
```

File: extensions/markdown-reference/src/markdown-escape.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const replacements: Array<Array<RegExp | string>> = [
+  [/\*/g, "\\*", "asterisks"],
+  [/#/g, "\\#", "number signs"],
+  [/\//g, "\\/", "slashes"],
+  [/\(/g, "\\(", "parentheses"],
+  [/\)/g, "\\)", "parentheses"],
+  [/\[/g, "\\[", "square brackets"],
+  [/\]/g, "\\]", "square brackets"],
```
Comment (pernielsentikaer) [inline extensions/markdown-reference/src/markdown-escape.ts:8]:
```suggestion
  [/]/g, "\\]", "square brackets"],
```
Removed unnecessary escape

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @codedbyjordan 👋 

Congratulations on your first contribution here, awesome extensions!

I left some comments for you to look into 🔥 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by codedbyjordan** [comment]
> Hi @codedbyjordan 👋
> 
> Congratulations on your first contribution here, awesome extensions!
> 
> I left some comments for you to look into 🔥
> 
> Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

Ready!

**Review by pernielsentikaer — APPROVED** [review]
hi @codedbyjordan 👋 

Awesome, thanks for the corrections 🔥 I'm going to accept it and then will my colleague do a final review before it's finally merged 🚢

**Comment by fe9lix** [comment]
Hey @codedbyjordan, one suggestion from my side: how about showing aa reference list items when opening the command? (The default "No results" right after opening the commands is not so nice for end users...)

**Comment by pernielsentikaer** [comment]
That makes totally sense, we should 100% do that 😄

**Comment by codedbyjordan** [comment]
> That makes totally sense, we should 100% do that 😄

Added!

**Comment by codedbyjordan** [comment]
> Hey @codedbyjordan, one suggestion from my side: how about showing a reference list items when opening the command? (The default "No results" right after opening the command is not so nice for end users...)

Good suggestion! Added that in :)

**Comment by fe9lix** [comment]
One more nit: "Markdown Reference - Search Markdown Syntax" as command title feels too long. I would shorten this to just "Search Markdown Syntax" (you could add  "Markdown Reference" as subtitle but it doesn't seem to add much to the command title so I would rather leave that empty).

**Comment by codedbyjordan** [comment]
> One more nit: "Markdown Reference - Search Markdown Syntax" as command title feels too long. I would shorten this to just "Search Markdown Syntax" (you could add "Markdown Reference" as subtitle but it doesn't seem to add much to the command title so I would rather leave that empty).

Got it :) Thank you !!

**Comment by pernielsentikaer** [comment]
Much better @codedbyjordan 🔥
